ipam/allocator_test.go

```diff
@@ -987,6 +976,122 @@ func TestRequest(t *testing.T) {
 	}
 }
 
+// TestOverlappingRequests tests that overlapping subnets cannot be allocated.
+// Requests for subnets which are supersets or subsets of existing allocations,
+// or which overlap at the beginning or end, should not be permitted.
+func TestOverlappingRequests(t *testing.T) {
+	input := []struct {
+		environment []string
+		subnet      string
+		ok          bool
+	}{
+		// Ok: Non-overlapping left and right
+		{[]string{"10.0.0.0/8"}, "11.0.0.0/8", true},
+		{[]string{"10.0.0.0/8"}, "9.0.0.0/8", true},
+		{[]string{"172.16.0.0/16"}, "172.15.0.0/16", true},
+		{[]string{"172.16.0.0/16"}, "172.17.0.0/16", true},
+
+		{[]string{"11.0.0.0/8"}, "10.0.0.0/8", true},
+		{[]string{"9.0.0.0/8"}, "10.0.0.0/8", true},
+		{[]string{"172.15.0.0/16"}, "172.16.0.0/16", true},
+		{[]string{"172.17.0.0/16"}, "172.16.0.0/16", true},
+
+		// Not ok: Supersets and subsets
+		{[]string{"10.0.0.0/8"}, "10.0.0.0/8", false}, // exact overlap
+		{[]string{"10.0.0.0/8"}, "10.0.0.0/7", false}, // superset
+		{[]string{"10.0.0.0/8"}, "10.0.0.0/6", false}, // superset, non-canonical
+		{[]string{"10.0.0.0/8"}, "8.0.0.0/6", false},  // superset, canonical
+		{[]string{"10.0.0.0/8"}, "10.0.0.0/9", false}, // subset
+
+		{[]string{"10.0.0.0/8"}, "10.0.0.0/16", false},
+		{[]string{"10.0.0.0/8"}, "10.0.0.0/24", false},
+
+		// IPv4
+		// Previously allocated network does not overlap with request
+		{[]string{"74.0.0.0/7"}, "9.111.99.72/30", true},
```

Comment:
how are these different from the "ok" section ?

Reply:
These are script-generated test cases which cover the same property.   I can combine them with the earlier ones and keep some of both - the 10.0.0.0 examples are easier to read and understand.